### DISCUSSION TITLE
v1.13 Release Candidate 1

### DISCUSSION THREAD

**Discussion by dorkster** [flareteam/flare-game#878] [Announcements]
## Engine features

* Added additional requirement properties (requires_status, requires_item, etc) to enemy and NPC map objects. The properties now match what is available for map events.
* When 'mouse aim' is disabled, the Shift key can now be used with the direction keys to change the player's orientation without moving.
* Added optional 'fog-of-war' feature (r-a-cristian-93)
* Added support for multiple sets/loadouts of equipped items (r-a-cristian-93)
* Added random_status attribute to Events for setting/unsetting a random status from a list.
* Reworked gamepad handling using the SDL_GameController API.
* Several new key bindings, many of which will benefit gamepad users.
* Added an "action picker" for interacting with items/powers when using a gamepad or a touchscreen.

## Engine fixes

* Fix enemies not being able to target NPC followers
* Fix loot and sound effects being cleared in StatBlock files with multiple sections
* Fix potential crash when removing summons that would be out-of-bounds on the next map.
* Fix bug where a damage type with no name could be removed if another damage type was defined after it.
* Don't show hidden stats under 'Related Stats' section of tooltips of Character menu.
* Fix bug where negative regen wouldn't work if the resource was 100% full.
* Fix bug where negative regen would overflow after reaching zero.
* Fix bug that caused built-in effect types to be ignored by Power post_effect.
* Fix disappearing sprites due to render priority overflow.
* Fading of corpses is now correctly disabled when corpse_timeout=0 is used in engine/misc.txt.
* Fix cast animation not playing when using an inventory item that is not instant.
* Fixed a bug on Android where loading the default settings would exit fullscreen mode.
* Fixed crash in NPC dialog when using gamepad.
* Fixed several small bugs with navigating UI elements with a gamepad.
* Fixed incorrect handling of gamepad stick/trigger deadzone.
* Fixed fullscreen setting being turned off when loading configuration defaults on Android.
* Fixed input hint for item tooltips when using a touchscreen.
* Improved file access handling on Android versions 10 and up. This fixes the infinite cutscene loop when starting a new game.
* Added misc.selection_rect_corner_size to engine/widget_settings.txt. This allows for a nicer "targeting reticle" appearance when selecting widgets without a mouse.

## Game updates

* New formula for the 'Shield' power. Instead of Mental Damage, it is now based off a combination of player level, Mental stat, and Defense stat. This should increase its usefulness on mixed builds.
* Added 'Super' and 'Ultra' variants of Health and Mana potions. They can be obtained through the Alchemy system.
* New icons for Health and Mana potions variants.
* Fixed Alchemist's Skirt so that it works with all potions.
* Prevent the player from damaging Mez before being trapped.
* Made Lezaith immune to slow/stun effects.
* Goblin Thief can now steal all available kinds of potions. Its drop table has also been expanded to make it easier to replenish lost potions.
* Improved contrast for "highlight" state of red square buttons ("close" and "plus").

## Translation updates

* Scottish Gaelic (gd) update (GunChleoc)
* Polish (pl) update (Pawonashi)

**Comment by Danimal696** [comment]
A question, can you expand on the use of these?:
- Added additional requirement properties to enemy and NPC map objects. Like what properties? is there a list somewhere?
- Added support for multiple equipment sets. Wasn´t there already?
- Added 'fog-of-war'. Not a fan of this, can it be disabled?
- Added random_status attribute to Events. Uh? totally lost on this one, needs explanation.

Sorry, but as the developer you know what  everything is, for someone like me it´s a total loss. Anyways, congrats on the new release ;)

**Reply by dorkster** [reply]
(Moving my comment because I forgot to make it a reply to yours.)

- "Requirement properties" are all the ones that start with `requires_`, including: `requires_status`, `requires_level`, `requires_currency`, `requires_item` and `requires_class`. This also includes all of the "not" variants of those properties. These all existed for events, so this addition is just bringing map enemies and NPCs up to parity.
- Equipment sets is not to be confused with item sets. An equipment sets provide a way to quickly swap between different loadouts of equipped items. For example, a player could have a secondary loadout focused on loot/XP grinding separate from their main loadout. The feature isn't currently used in the Empyrean campaign, but I might enable it since it's a nice quality-of-life feature. The original PR has some screenshots: https://github.com/flareteam/flare-engine/pull/1789
- Fog of war is entirely optional for mods to use. The Empyrean campaign won't use it. However, there's the option to use the fog data to gradually reveal *only* the minimap as it's explored without rendering any fog tiles. I might consider using that, since it's not as obtrusive. Original PR: https://github.com/flareteam/flare-engine/pull/1780
- The `random_status` attribute is a way to create a list of statuses and then pick one randomly from the list. For an example, have a look at the updated Goblin Thief script: https://github.com/flareteam/flare-game/blob/master/mods/empyrean_campaign/scripts/powers/goblin_steal.txt

**I've updated the release notes to make these things more clear.**
